mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/model/FlagsFilterTest.java

```diff
@@ -0,0 +1,63 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.mailbox.store.mail.model;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.List;
+
+import javax.mail.Flags;
+
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableList;
+
+public class FlagsFilterTest {
+
+    private List<Flags.Flag> listOfFlags = ImmutableList.of(Flags.Flag.SEEN, Flags.Flag.RECENT, Flags.Flag.FLAGGED);
+    private List<String> listOfUserFlags = ImmutableList.of("VeryImportant", "Bof");
+
+    @Test
+    public void buildShouldNotRequireAnyProperty() {
+        assertThat(FlagsFilter.builder().build()).isNotNull();
+    }
+
+    @Test
+    public void buildWithoutPropertyShouldReturnNoFilter() {
+        FlagsFilter filter = FlagsFilter.builder().build();
+        assertThat(listOfFlags.stream().filter(filter.getSystemFlagFilter())).isEqualTo(listOfFlags);
+        assertThat(listOfUserFlags.stream().filter(filter.getUserFlagFilter())).isEqualTo(listOfUserFlags);
+    }
+
+    @Test
+    public void buildWithSystemFlagFilterShouldNotFilterUserFlags() {
+        FlagsFilter filter = FlagsFilter.builder()
+            .systemFlagFilter(flag -> false)
+            .build();
+        assertThat(listOfUserFlags.stream().filter(filter.getUserFlagFilter())).isEqualTo(listOfUserFlags);
+    }
+
+    @Test
+    public void buildWithUSerFlagFilterShouldNotFilterSystemFlags() {
+        FlagsFilter filter = FlagsFilter.builder()
+            .userFlagFilter(flag -> false)
+            .build();
+        assertThat(listOfFlags.stream().filter(filter.getSystemFlagFilter())).isEqualTo(listOfFlags);
+    }
```

Comment:
missing buildWithBoth also

Reply:
I just supposed that putting two objects in my object was ok. the test would basically be setObjet -> build -> getObject.
I don't think it's worth if.
WDYT ?

Reply:
it would cover the case where one filter overwrite the other (not really possible with this implementation, but it could change)

Reply:
done